SConstruct

```diff
@@ -253,6 +253,7 @@ condition_map = {
     'ARM64_HOST' : False,
     'POSIX_HOST' : env['PLATFORM'] == 'posix',
     'POWERPC_HOST' : False,
+    'RISCV64_HOST' : False,
```

Comment:
If `SConstruct` is really only for Windows builds, this could probably be dropped, but I did see that there was support here for the other architectures.

Reply:
Yes; SConstruct does have some limited support for platforms other than Windows, probably because at the time, there was an expectation we'd switch to SCons as the WiredTiger build platform. Since CMake was chosen, I don't know what the deal is, but my guess is that the right approach is to strip everything not related to Windows from SConstruct.  @alisonfel, you know more about this than I do, so I'll defer to you on whether this needs a cleanup ticket.

Reply:
There's also the whole "build checksum stuff as a separate library", which I've kind of lost track of, but yeah, I'm good with whatever @alisonfel decides.